### ISSUE TITLE
Screen readers cannot announce checked state of checkable ToolStrip/MenuStrip menu items 

### ISSUE THREAD

**Issue by dmitrii-drobotov** [dotnet/winforms#8307]
### .NET version

.NET 8

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, but it was working in [MainMenu](https://learn.microsoft.com/dotnet/api/system.windows.forms.mainmenu) control

### Issue description

Initially reported in https://github.com/dotnet/winforms/issues/7770. Screen readers cannot announce checked state of checkable ToolStrip/MenuStrip menu items due to lack of [Toggle pattern](https://learn.microsoft.com/en-us/dotnet/framework/ui-automation/implementing-the-ui-automation-toggle-control-pattern) implementation.

### Steps to reproduce

1. Add MenuStrip to form, initialize it with menu items (for example via "Insert standard items" button)
2. Make items checkable: set `CheckOnClick` property to true or `CheckState` property to `Checked`
3. Start Narrator and run the app
5. Focus on MenuStrip, select checked menu item — Narrator will not announce that item is checked

Sample app:
[Uploading WinFormsMenuStripChecked.zip…]()

**Comment by Olina-Zhang** [comment]
@dmitrii-drobotov I cannot repro this issue, Narrator can announce the checked state for checkable ToolStripMenuItem:
![image](https://user-images.githubusercontent.com/26474449/205570344-033575c9-540d-4156-ba9b-b355ec312fdb.png)

**Comment by dmitrii-drobotov** [comment]
> @dmitrii-drobotov I cannot repro this issue, Narrator can announce the checked state for checkable ToolStripMenuItem: 

@Olina-Zhang - looks like Narrator announced it in Win 10 and didn't announce in Win 11, that's why it reproduced for me. But in this issue we mostly wanted to fix it in JAWS and NVDA, as reported in #7770, where it didn't work at all.

**Comment by MandiMan** [comment]
Verified this issue on .NET latest build: 8.0.100-alpha.1.23055.16, issue was fixed that Narrator could announces checked state of checkable menu item.
![issue8307 win10](https://user-images.githubusercontent.com/108860782/210953611-e2540115-4588-4899-941a-e242b0c67327.png)
